modules/top-level/test.nix

```diff
@@ -19,23 +90,79 @@ in
       description = "The test derivation's name.";
     };
 
+    buildNixvim = lib.mkOption {
+      type = lib.types.bool;
+      description = "Whether to build the nixvim config in the test.";
+      default = true;
+    };
+
     runNvim = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to run `nvim` in the test.";
-      default = true;
+      defaultText = lib.literalExpression "config.test.buildNixvim";
+      default = cfg.buildNixvim;
     };
 
     checkWarnings = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to check `config.warnings` in the test.";
+      apply = x: lib.warnIfNot x "`test.checkWarnings = false` is replaced with `test.warnings = [ ]`." x;
       default = true;
     };
 
     checkAssertions = lib.mkOption {
       type = lib.types.bool;
       description = "Whether to check `config.assertions` in the test.";
+      apply =
+        x: lib.warnIfNot x "`test.checkAssertions = false` is replaced with `test.assertions = [ ]`." x;
       default = true;
     };
+
+    warnings = lib.mkOption {
+      type = lib.types.listOf expectationType;
+      description = "A list of expectations for `warnings`.";
+      defaultText = lib.literalMD "Expect `count == 0`";
+      default = [ 0 ];
+    };
+
+    assertions = lib.mkOption {
+      type = lib.types.listOf expectationType;
+      description = "A list of expectations for `assertions`.";
+      defaultText = lib.literalMD "Expect `count == 0`";
+      default = [ 0 ];
+    };
+
+    predefinedExpectationPredicates = lib.mkOption {
+      type =
+        with lib.types;
+        attrsOf (submodule {
+          options = {
+            predicate = lib.mkOption {
+              type = functionTo (functionTo bool);
+              description = ''
+                Predicate matching `(value) -> [(message)] -> Boolean`.
+              '';
+            };
+            message = lib.mkOption {
+              type = functionTo (either str (functionTo str));
+              description = ''
+                Expectation message supplier, matching `(value) -> String` or `(value) -> [(message)] -> String`.
+              '';
+            };
+            valueType = lib.mkOption {
+              type = lib.types.optionType;
+              description = ''
+                The type to use for `value` when this expectation is used.
+              '';
+              default = lib.types.str;
```

Comment:
Not sure if I agree with specifying a default type for this, I don't really see why a `str` is a "default" type for an assertion value

Reply:
This was the default because it is the most commonly use. As it's an internal option I didn't think it was too important, but you're right it should probably be explicit.